rct229/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/do_all_terminals_have_one_fan.py

```diff
@@ -0,0 +1,31 @@
+from rct229.ruleset_functions.baseline_systems.baseline_system_util import (
+    find_exactly_one_terminal_unit,
+)
+
+
+def do_all_terminals_have_one_fan(rmi_b, terminal_unit_id_list):
+    """Returns TRUE if the fan data element associated with all terminal units input to this function are equal to one (i.e., there is only one fan associated with the terminal unit). It returns FALSE if any terminal unit has a fan data element not equal to one (i.e., there is NOT only one fan associated with the terminal unit).
+
+    Parameters
+    ----------
+    rmi_b: json
+        RMD at RuleSetModelInstance level
+    terminal_unit_id_list: list
+        List of terminal_units IDs
+
+    Returns
+    -------
+    bool
+        True: the fan data element associated with all terminal units input to this function are equal to one (i.e., there is only one fan associated with the terminal unit).
+        False: any terminal unit has a fan data element not equal to one (i.e., there is NOT only one fan associated with the terminal unit).
+    """
+
+    do_all_terminals_have_one_fan = True
+
+    for terminal_b_id in terminal_unit_id_list:
+        terminal_b = find_exactly_one_terminal_unit(rmi_b, terminal_b_id)
+        if terminal_b.get("fan") == None:
+            do_all_terminals_have_one_fan = False
+            break
+
+    return do_all_terminals_have_one_fan
```

Comment:
Great point! 

I completely agree that the suggestion is a more succinct and pythonic way. 

However, if we think about the code performance, wouldn't the original code be more efficient? Because in the original code, the loop stops as soon as the ```fan``` key doesn't exist. However, the suggested code iterates all the ```fan``` key elements.

I'd appreciate it you let me know what you think about this!

Reply:
It is excellent that you have started thinking about coding performance and styles.
To your question, I want to mention two things - Charlie might be able to contribute more: 

1. In Python, all() function does short circuit, so if the condition is false, all will immediately stop the loop and return `False`
2. Normally, code performance is evaluated by Big O. In most scenarios, a loop is O(n), where n is the number of iterations. In this representation, the loop, whether it breaks or continues, the code performance is O(n). So there is no difference if we put a break condition or not.

Reply:
Thank you so much for this explanation Weili. This is wonderful!

I just addressed the comment :).